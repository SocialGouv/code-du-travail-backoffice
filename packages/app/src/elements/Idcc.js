import styled from "@emotion/styled";
import React from "react";

import { validateMandatoryNullableString } from "../props/validators";

const Container = styled.div`
  background-color: var(--color-label-background);
  border: solid 1px var(--color-border);
  border-radius: 0.25rem;
  cursor: help;
  font-size: 0.8rem;
  font-weight: 600;
  margin-right: 0.75rem;
  max-width: 3rem;
  min-width: 3rem;
  opacity: 0.75;
  padding: 0.25rem 0;
  position: relative;
  text-align: center;
  user-select: none;
  width: 3rem;
  white-space: nowrap;

  > div {
    background-color: var(--color-eerie-black);
    border-radius: 0.25rem;
    box-shadow: 0 0 0.5rem var(--color-eerie-black);
    color: white;
    display: none;
    font-size: 0.8rem;
    font-weight: 600;
    left: -1px;
    padding: 0.35rem 0.6rem 0.4rem;
    position: absolute;
    text-align: left;
    bottom: 2rem;
    width: 22rem;
    white-space: normal;
    z-index: 2;
  }

  :hover {
    opacity: 1;

    > div {
      display: block;
    }
  }
`;

const Idcc = ({ code, name, ...props }) => (
  <Container {...props}>
    {code === null ? "CDT" : code}
    <div data-testid="tooltip">{name === null ? "Code du travail" : name}</div>
  </Container>
);

Idcc.propTypes = {
  code: validateMandatoryNullableString,
  name: validateMandatoryNullableString,
};

export default Idcc;
