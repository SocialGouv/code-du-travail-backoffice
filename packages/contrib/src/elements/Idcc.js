import styled from "@emotion/styled";
import React from "react";
import ReactTooltip from "react-tooltip";

const Container = styled.div`
  background-color: var(--color-label-background);
  border: solid 1px var(--color-black-leather-jacket);
  border-radius: 1rem;
  cursor: help;
  font-size: 0.8rem;
  font-weight: 600;
  margin-right: 0.75rem;
  opacity: 0.75;
  padding: 0.25rem 0.5rem;
  user-select: none;
  white-space: nowrap;

  :hover {
    opacity: 1;
  }
`;

const Tooltip = styled(ReactTooltip)`
  max-width: 22rem;
  padding: 0.25rem 0.5rem 0.45rem;
  white-space: normal;
`;

const Idcc = ({ code, name, ...props }) => (
  <Container data-tip={code === undefined ? "Code du travail" : name} {...props}>
    {code === undefined ? "CDT" : `IDCC: ${code}`}
    <Tooltip />
  </Container>
);

export default Idcc;
