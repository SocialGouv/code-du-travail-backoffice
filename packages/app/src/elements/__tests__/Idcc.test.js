import React from "react";

import Idcc from "../Idcc";

// TODO Unskip this tests once "react-tooltip" is fixed.
// https://github.com/wwayne/react-tooltip/issues/562
describe.skip("elements/<Idcc />", () => {
  const COMMON_PROPS = {
    name: "A Labor Agreement",
  };

  it(`should pass`, () => {
    const $idcc = testRender(<Idcc {...COMMON_PROPS} />);

    expect($idcc).toMatchSnapshot();
    expect($idcc.props).toHaveProperty("data-tip", "Code du travail");
    expect($idcc.children[0]).toStrictEqual("CDT");
  });

  it(`should pass with {code} = 1234`, () => {
    const CODE = 1234;
    const $idcc = testRender(<Idcc code={CODE} {...COMMON_PROPS} />);

    expect($idcc.props).toHaveProperty("data-tip", COMMON_PROPS.name);
    expect($idcc.children[0]).toStrictEqual(`IDCC: ${CODE}`);
  });
});
