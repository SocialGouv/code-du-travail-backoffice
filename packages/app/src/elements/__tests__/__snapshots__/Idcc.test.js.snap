// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/<Idcc /> should pass 1`] = `
.emotion-0 {
  background-color: var(--color-label-background);
  border: solid 1px var(--color-black-leather-jacket);
  border-radius: 1rem;
  cursor: help;
  font-size: 0.8rem;
  font-weight: 600;
  margin-right: 0.75rem;
  opacity: 0.75;
  padding: 0.25rem 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
}

.emotion-0:hover {
  opacity: 1;
}

<div
  className="emotion-0 emotion-1"
  data-tip="Code du travail"
>
  CDT
  <div
    className="__react_component_tooltip place-top type-dark"
    data-id="tooltip"
  />
</div>
`;
